### ISSUE TITLE
Add versioning to processors

### ISSUE THREAD

**Issue by JeanMertz** [blendle/automaat#4]
Processors are pre-defined sets of code combined with a set of configuration options. The name of a processor, combined with its options is serialised and stored into the database when a pipeline is created, then upon running that pipeline, the stored data is deserialised back into an actual processor that can be executed.

The downside to this design is that deserialising into the original processor might fail, if the type signature of that processor changed after it was serialised to the database.

To mitigate this, a processor should be versioned, and that version should be part of the serialised information stored in the database.

Any change to a processor that breaks deserialisation (meaning, a change to the type signature) would require a version bump. Technically, runtime changes (so, what a processor actually _does_ when it runs) is not a breaking change for deserialising, but that too has to be handled with care, in order to not break existing pipelines.

There's still some thinking to do on how this impacts the API, as this would also have to be conveyed somehow via GraphQL.

Until this is solved, whenever a processor is changed in a backward breaking way, a query has to be executed to update all records that serialised this processor, to modify the data to make it deserialisable again.

So, until this is solved, let's keep backward breaking changes to a minimum 😅

**Comment by JeanMertz** [comment]
I've split off all processors that currently exist into their own crates:

* https://docs.rs/automaat-processor-git-clone
* https://docs.rs/automaat-processor-print-output
* https://docs.rs/automaat-processor-redis-command
* https://docs.rs/automaat-processor-shell-command
* https://docs.rs/automaat-processor-string-regex

So each processor is now versioned according to the SemVer spec.

Given that Cargo allows including multiple versions of the same package, this means we can have (f.e.) a `GitCloneV1` object and a `GitCloneV2` object. We still need to do some work to serialize this information into the database, but once that's done (and as long as we adhere to the SemVer spec), I think this issue is actually solved.

**Comment by JeanMertz** [comment]
Closing. This is solved, as we can add v2 packages when needed, and give them different GraphQL object names (probably just postfixed with `V2`, as mentioned above). I've opted to use the v1 packages without the `V1` prefix, as we might never need to release a V2, and this makes using these objects a bit nicer.

See:

https://github.com/blendle/automaat/blob/51c252cc362b657bbfc1738aa6b3970f9d7d0319/src/server/Cargo.toml#L50-L96

https://github.com/blendle/automaat/blob/51c252cc362b657bbfc1738aa6b3970f9d7d0319/src/server/src/processor.rs#L107-L122
